parameter/EnumParameterType.cpp

```diff
@@ -129,20 +131,46 @@ int32_t CEnumParameterType::getMax() const
 }
 
 bool CEnumParameterType::fromBlackboard(string &userValue, const uint32_t &value,
-                                        CParameterAccessContext & /*ctx*/) const
+                                        CParameterAccessContext &ctx) const
 {
     // Convert the raw value from the blackboard
     int32_t signedValue = static_cast<int32_t>(value);
     signExtend(signedValue);
 
-    // Convert from numerical space to literal space
-    return getLiteral(signedValue, userValue);
+    // Take care of format
+    if (ctx.valueSpaceIsRaw()) {
+
+        // Format
+        std::ostringstream sstream;
+
+        // Numerical format requested
+        if (ctx.outputRawFormatIsHex()) {
+
+            // Hexa display with unecessary bits cleared out
+            sstream << "0x" << std::hex << std::uppercase
+                    << std::setw(static_cast<int>(getSize() * 2)) << std::setfill('0')
```

Comment:
pretty sure this static cast is not useful. Or maybe only to go from unsigned to signed ?